### PR TITLE
Refactor all classes and tests

### PR DESCRIPTION
**Pull request by suzkiee**
## What does this change?

board class  
game class
turn class
setup class  
ship spec

## What does this fix?

board class - test generate cells method and helper methods
game class - update messages with spaces
turn class - test take_turn method and validate_computer_shot method 
setup class  -  update test for run_setup
ship spec - update spec ship name to "rocket" 🚀

## Is this a bug or a feature? 
Feature updates

## How was this tested?
See above.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -6,75 +6,40 @@ def initialize
...
-  def numbers(placement_coordinates)
-    placement_coordinates.map do |coordinate|
-      coordinate.split('').last.to_i
-    end
-  end
-
-  def consecutive?(array)
-    array.each_cons(2).all? do |num_1, num_2|
-      num_2 == num_1 +1
+  def create_keys(letters, numbers)
```
Comment (mkrumholz) [inline lib/board.rb:36]: Yes, love this whole refactor

File: spec/board_spec.rb [hunk 2]
```diff
@@ -11,11 +11,13 @@
       board = Board.new
       expect(board).is_a? Board
     end
+  end
 
+  describe '#generate_cells' do
```
Comment (mkrumholz) [inline spec/board_spec.rb:16]: Great to test this! We should replace any "placement_coordinates" instances with "coordinates" here

File: lib/turn.rb [hunk 3]
```diff
@@ -28,7 +28,7 @@ def user_shoots(computer)
   end
 
   def validate_user_shot(computer, coordinate)
-    while invalid?(computer, coordinate)  || already_fired_on?(computer, coordinate)
+    while invalid?(computer, coordinate) || already_fired_on?(computer, coordinate)
```
Comment (mkrumholz) [inline lib/turn.rb:31]: Way cleaner.

File: spec/turn_spec.rb [hunk 4]
```diff
@@ -31,11 +30,9 @@
       ships = [submarine, cruiser]
       computer_setup = Setup.new(ships, board, :computer)
       computer = computer_setup.run_setup
+      mock_user = instance_double("Board")
 
-      # user_setup = Setup.new(ships, board, :human)
-      # user = user_setup.run_setup
-
-      turn = Turn.new(user, computer, player_type)
+      turn = Turn.new(mock_user, computer, :computer)
```
Comment (mkrumholz) [inline spec/turn_spec.rb:35]: Woo for mocks!

File: spec/turn_spec.rb [hunk 5]
```diff
@@ -46,14 +43,36 @@
...
+      mock_user = instance_double("Board")
 
-      turn = Turn.new(user, computer, player_type)
+      turn = Turn.new(mock_user, computer, :computer)
 
       expect(turn.player_type).to eq :computer
     end
   end
 
+  describe '#take_turn' do
```
Comment (mkrumholz) [inline spec/turn_spec.rb:55]: Super good that we could test this method. Huzzah!

File: spec/board_spec.rb (hunk 2 above)
Comment (suzkiee) [reply spec/board_spec.rb:16]: yeahh those need to go

**Pull request merged**